test/unit/index.js

```diff
@@ -155,47 +161,121 @@ describe('extractData()', function () {
                 expect(extractData).withArgs({locales: ['foo-bar']}).to.throwException();
             });
 
+            it('should always contribute an entry for all specified `locales`', function () {
+                var data = extractData({
+                    locales       : ['en-US', 'zh-Hans-SG'],
+                    pluralRules   : true,
+                    relativeFields: true,
+                });
+
+                expect(data).to.have.keys('en-US', 'zh-Hans-SG');
+            });
+
             it('should recursively expand `locales` to their roots', function () {
                 var data = extractData({
-                    locales       : ['en-US', 'zh-Hant-TW'],
+                    locales       : ['en-US', 'zh-Hans-SG'],
                     pluralRules   : true,
                     relativeFields: true,
                 });
 
-                expect(data).to.have.keys('en', 'zh', 'zh-Hant');
+                expect(data).to.have.keys('en', 'zh', 'zh-Hans');
             });
 
             it('should accept `locales` of any case and normalize them', function () {
                 expect(extractData({
                     locales     : ['en-us', 'ZH-HANT-HK'],
                     pluralRules : true,
-                })).to.have.keys('en', 'zh');
-
-                expect(extractData({
-                    locales       : ['en-us', 'ZH-HANT-HK'],
-                    pluralRules   : true,
-                    relativeFields: true,
-                })).to.have.keys('en', 'zh', 'zh-Hant', 'zh-Hant-HK');
```

Comment:
Removed test for exceptional case. The old check was incorrect since `zh` won't ever be outputted since `zh-Hant`'s parent is actually the "root" locale.